docs/basics/101-146-providers.rst

```diff
@@ -0,0 +1,86 @@
+.. _providers:
+
+Configuring new data providers
+------------------------------
```

Comment:
I know that this is the term used in the code, but IMHO what is being specified is the access method to a given resource. I believe that any code could be executed here given some kind of matching URL. Maybe it is just me, but the term "provider" feels off. I wonder whether we should make prominent use of it. The only two places that term appears are in the filename of `.datalad/providers/....cfg` and in a single config section header in such a file `[provider:....]`.

Maybe just "configure custom data access"?